### PR TITLE
fix(payment): CHECKOUT-6967 Fix issue with icon not highlighting for stored instruments

### PR DESCRIPTION
**Pull request by animesh1987**
## What?
- Fix issue with card icon not highlighting for stored instruments
- Use https://github.com/bigcommerce/checkout-js/pull/1295 as a reference to move tests

## Why?
Selecting a stored instrument method on checkout doesn't update the selected credit card icon, this leads to an inconsistent UX.

## Testing / Proof
When stored card is selected
Screenshot 2023-07-20 at 9 13 18 pm

User enters card number
Screenshot 2023-07-20 at 9 15 26 pm

@bigcommerce/checkout

### CODE REVIEW

File: packages/core/src/app/payment/paymentMethod/PaymentMethodTitle.tsx [hunk 1]
```diff
@@ -282,7 +300,7 @@ const PaymentMethodTitle: FunctionComponent<
                 )}
 
                 {titleText && (
-                    <div className="paymentProviderHeader-name" data-test="payment-method-name">
+                    <div aria-level={6} className="paymentProviderHeader-name" data-test="payment-method-name" role="heading">
```
Comment (bc-peng) [inline packages/core/src/app/payment/paymentMethod/PaymentMethodTitle.tsx:303]: I didn't find any other `aria-level` reference in the codebase.
Can you please explain why `6` is approriate here?

File: packages/core/src/app/payment/paymentMethod/PaymentMethodTitle.test.tsx [hunk 2]
```diff
@@ -0,0 +1,719 @@
...
+                    }}
+                />,
+            );
+
+            const logos = screen.getByRole('img');
+            const names = screen.getByRole('heading');
+
+            expect(logos).toBeInTheDocument();
+
+            expect(names).toBeInTheDocument();
```
Comment (bc-peng) [inline packages/core/src/app/payment/paymentMethod/PaymentMethodTitle.test.tsx:193]:
```suggestion
            expect(screen.getByRole('img')).toBeInTheDocument();
            expect(screen.getByRole('heading')).toBeInTheDocument();
```

File: packages/core/src/app/payment/paymentMethod/PaymentMethodTitle.tsx (hunk 1 above)
Comment (animesh1987) [reply packages/core/src/app/payment/paymentMethod/PaymentMethodTitle.tsx:303]: Just wanted to stick to a level that's not in top so a screen reader thinks this is a main header.

File: packages/core/src/app/payment/paymentMethod/PaymentMethodTitle.test.tsx [hunk 3]
```diff
@@ -0,0 +1,668 @@
...
+                    }}
+                />,
+            );
+
+            const logos = screen.getByRole('img');
+            const names = screen.queryByRole('heading');
+
+            expect(logos).toBeInTheDocument();
+
+            expect(names).not.toBeInTheDocument();
```
Comment (bc-peng) [inline packages/core/src/app/payment/paymentMethod/PaymentMethodTitle.test.tsx:391]:
```suggestion
            expect(screen.getByRole('img')).toBeInTheDocument();
            expect(screen.queryByRole('heading')).not.toBeInTheDocument();
```

One place left.

**Review by bc-peng — APPROVED** [review]
LGTM except for a minor issue. 👏

**Pull request merged**